File: features/steps/google_home_page_steps.rb
require "selenium/client"

Given /^that I am on the browser start page$/ do
  @browser = Selenium::Client::Driver.new :host => "localhost", :port => 4444, :browser => "*firefox", :url => "http://www.google.com.br", :timeout_in_second => 60
  @browser.start_new_browser_session
end

When /^I go to the website www\.google\.com\.br$/ do
  @browser.open("http://www.google.com.br")
end

Then /^the title of page should be "([^"]*)"$/ do |title|
  assert_equal @browser.title, title
  @browser.close_current_browser_session
end

File: src/net/franciscosouza/steps/GoogleHomePageSteps.java
package net.franciscosouza.steps;

import static cuke4duke.annotation.I18n.EN.*;
import com.thoughtworks.selenium.SeleneseTestCase;

import cuke4duke.annotation.After;
import cuke4duke.annotation.Before;

public class GoogleHomePageSteps extends SeleneseTestCase {
    
    @Before()
    public void setUpSelenium() throws Exception {
        this.setUp("http://www.google.com.br", "*chrome");
    }
    
    @Given("that I am on the browser start page")
    public void openBrowser() {
        this.selenium.open("/");
    }
    
    @When("I go to the website www.google.com.br")
    public void goToGoogle() {
        this.selenium.open("http://www.google.com.br");
    }
    
    @Then("the title of page should be \"(.*)\"")
    public void checkTitle(String title) {
        assertEquals(title, this.selenium.getTitle());
    }
    
    @After
    public void closeSelenium() {
        this.selenium.close();
    }
    
}
